ComponentKit/Utilities/CKParam.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <initializer_list>
+#include <vector>
+#include <unordered_map>
+
+namespace CK {
+  /**
+   * Helper class that's useful for function parameters that's likely to take
+   * initializer lists as input. It prevents inlined vector construction.
+   *
+   * For a function of the form: `foo(const vector<int> &)`,
+   *   invoked with an initializer list: `foo({1,2,3})`,
+   *   code is generated to construct and destroy a vector from that initializer
+   *   list, usually inlined into the body of the function.
+   * For a function of the form: `foo(VectorParam<int> &&)`,
+   *   invoked with an initializer list: `foo({1,2,3})`,
+   *   the vector constructor and destructor are not inlined. This lack of
+   *   inlining helps reduce the size of the compiled output.
+   *
+   * Additionally, this type allows vector to be moved in, as if the parameter
+   *   has a rvalue-reference overload (e.g. `foo(vector<int> &&)`). In ObjC++
+   *   methods, where overloads are not allowed, this gives the benefit of a
+   *   rvalue-reference overload at the cost of an additional move operation.
+   */
+  template<class T>
+  class VectorParam {
+    typedef std::vector<T> ContainerType;
+  public:
+    VectorParam() {}
+    VectorParam(std::initializer_list<typename ContainerType::value_type> items) : _value(items) {}
+    VectorParam(ContainerType &&items) : _value(std::move(items)) {}
+    VectorParam(const ContainerType &items) : _value(items) {}
+    VectorParam(const VectorParam<T> &) = delete;
```

Comment:
I'm a little fuzzy on the exact rules here, despite just having read them at http://en.cppreference.com/w/cpp/language/move_assignment . Do you have to explicitly default the move assignment operator because of this deleted copy ctor?